kedro-datasets/tests/pandas/test_sql_dataset.py

```diff
@@ -417,3 +418,56 @@ def test_create_connection_only_once(self, mocker):
         assert mock_engine.call_count == 2
         assert fourth.engines == first.engines
         assert len(first.engines) == 2
+
+    def test_adapt_mssql_date_params_called(self, mocker):
+        """Test that the adapt_mssql_date_params
+        function is called when mssql backend is used.
+        """
+        mock_adapt_mssql_date_params = mocker.patch(
+            "kedro_datasets.pandas.sql_dataset.SQLQueryDataSet.adapt_mssql_date_params"
+        )
+        mock_engine = mocker.patch("kedro_datasets.pandas.sql_dataset.create_engine")
+        ds = SQLQueryDataSet(sql=SQL_QUERY, credentials=dict(con=MSSQL_CONNECTION))
```

Comment:
```suggestion
        ds = SQLQueryDataSet(sql=SQL_QUERY, credentials={"con":MSSQL_CONNECTION})
```

Reply:
Oh yes, I saw the linter complaining as well but thought it wasn't a big deal. :smile: 
Will fix this!

Reply:
That being said, many more similar patterns exist in the code base but I won't fix them in this MR. @AhdraMeraliQB